src/sql/src/catalog.rs

```diff
@@ -103,7 +103,14 @@ pub trait Catalog: fmt::Debug {
     fn get_item_by_id(&self, id: &GlobalId) -> &dyn CatalogItem;
 
     /// Reports whether the specified type exists in the catalog.
-    fn type_exists(&self, name: &FullName) -> bool;
+    fn item_exists(&self, name: &FullName) -> bool;
+
+    /// Returns a lossy `ScalarType` associated with `id` if one exists.
+    ///
+    /// For example `pg_catalog.numeric` returns `ScalarType::Decimai(0,0)`,
+    /// meaning that its precision and scale need to be associated with values
+    /// from elsewhere.
+    fn try_get_scalar_type_by_name(&self, name: &PartialName) -> Option<ScalarType>;
```

Comment:
I think this comment is great! I'm lightly concerned that in a few months, etc. this function will be used by someone who doesn't read the disclaimer. Maybe we could rename this to something like `try_get_lossy_scalar_type_by_name`? (`lossy` might be enough to get people to think twice?)

Reply:
Sure thing